### PR TITLE
[BEAM-6632] Fixes integer overflow for interval >= 25 DAY

### PR DESCRIPTION
**Pull request by LesTR**
Fixed interger oveflow for fixed window interval with interval value >= 25

@akedin @apilloud @amaliujia Guys, can you please take a look on this? Thanks
------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [X ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [X ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by LesTR** [comment]
retest this please

**Comment by amaliujia** [comment]
LGTM

There was indeed an integer overflow.

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java [hunk 1]
```diff
@@ -181,12 +181,12 @@ private static RelNode updateWindow(RelOptRuleCall call, Aggregate aggregate, Pr
...
   private static Duration durationParameter(List<RexNode> parameters, int parameterIndex) {
-    return Duration.millis(intValue(parameters.get(parameterIndex)));
+    return Duration.millis(longValue(parameters.get(parameterIndex)));
   }
 
-  private static long intValue(RexNode operand) {
+  private static long longValue(RexNode operand) {
     if (operand instanceof RexLiteral) {
-      return RexLiteral.intValue(operand);
+      return ((Number) RexLiteral.value(operand)).longValue();
```
Comment (kennknowles) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java]: I also bet that adding this to Calcite at https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rex/RexLiteral.java#L1093 would be welcome.

**Comment by amaliujia** [comment]
Run Java PreCommit

**Comment by amaliujia** [comment]
Run JavaPortabilityApi PreCommit

**Comment by amaliujia** [comment]
master branch if failing. dev@ is discussing this issue. You will need a rebase after master branch is fixed to pass tests and get this PR merged.

**Comment by LesTR** [comment]
There is still one failing [test](https://builds.apache.org/job/beam_PreCommit_Java_Commit/4148/testReport/org.apache.beam.sdk.extensions.sql/BeamSqlDslAggregationTest/testTumbleWindowWith31DaysBounded/) Please don't merge it, i'll fix it today.

**Comment by XuMingmin** [comment]
@LesTR Got u, will wait for that.

**Comment by LesTR** [comment]
Run Java PreCommit

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java (hunk 1 above)
Comment (LesTR) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java]: @kennknowles filed CALCITE-2837 with patch

**Comment by LesTR** [comment]
@XuMingmin test fixed.

**Comment by LesTR** [comment]
Run Java PreCommit

**Comment by LesTR** [comment]
Run Java PreCommit

**Comment by robertwb** [comment]
Is this a change we should cherry-pick into the LTS branch?

**Comment by LesTR** [comment]
@robertwb by LTS you mean 2.7.x right? Sorry I am new in Beam. SQL didn't change for some time so i'll give +1 to this cherry-pick. It will be also awesome if this can be included in next release (2.10). It's simple change but maybe is too late for that.

**Comment by robertwb** [comment]
Yes, 2.7.x.

**Comment by kennknowles** [comment]
@LesTR the 2.10.0 release vote has just been finished, so further changes will make it into 2.11.0. You can find the "cut date" when we start working on each release on the [calendar](https://calendar.google.com/calendar/embed?src=0p73sl034k80oob7seouanigd0%40group.calendar.google.com) linked from https://beam.apache.org/community/policies/. The "good" news is that we will cut the release branch for 2.11.0 in two days. With luck, it will be released much faster so you'll have this change in a release quite soon.

**Comment by kennknowles** [comment]
(in fact, just as soon as we could build and vote on release a new RC for 2.10.0)

**Comment by LesTR** [comment]
@kennknowles thats fine. Thank You!

**Pull request merged**